src/metabase/lib/schema/ref.cljc

```diff
@@ -10,15 +11,16 @@
 (comment metabase.types/keep-me)
 
 (mr/def ::field.options
-  [:and
+  [:merge
    ::common/options
    [:map
+    [:base-type {:optional true} [:fn {:gen/gen (gen/elements (descendants :type/*))} #(isa? % :type/*)]]
```

Comment:
This seems like it should rather be `[:base-type {:optional true} (into [:enum] (descendants :type/*))]` - then wouldn't the generator work? Also there's `{:gen/elements ...}` which seems like a shorter road to this same semantics.

Reply:
I wanted to keep this definition from depending on the order the namespaces are loaded.